### PR TITLE
`Text exercises`: Fix feedback visualization for grading instructions with 0 points

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).

### Motivation and Context
This error was discovered by @willergeorg while testing #4756. When using grading instructions with 0 points, the feedback for the student is not displayed correctly (see Screenshot). 

### Description
The logic for determining whether or not a feedback block is empty did not take grading instructions into account. When using grading instructions, the feedback can have text even without adding another detail text. The newly added hasContent()-method allows the check for this case. 

### Steps for Testing
Prerequisites:
- 1 Exam with a text exerise
- 1 course text exercise
- Both exercises need grading instructions with 0 points

1. Participate in the exercises
2. Assess the submission using the grading instructions.
3. View the result as a student. Make sure that every feedback has a correct background color.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
old:
grafik

new:
grafik

### CODE REVIEW

**Review by Bofan-Zhu — DISMISSED** [review]
Tested on ts2. Text exercises in exam mode as well as normal ones work display the correct color 👍. 
image

**Review by dveljacic — APPROVED** [review]
Tested on TS2. Feedbacks have been correctly colored. ✅

File: src/main/webapp/app/entities/feedback.model.ts [hunk 1]
```diff
@@ -100,12 +100,17 @@ export class Feedback implements BaseEntity {
         return that.detailText != undefined && that.detailText.length > 0;
     }
 
+    public static hasContent(that: Feedback): boolean {
+        // if the feedback is associated with the grading instruction, the detail text is optional
+        return Feedback.hasDetailText(that) || !!(that.gradingInstruction && that.gradingInstruction.feedback);
```
Comment (bassner) [inline src/main/webapp/app/entities/feedback.model.ts:105]:
```suggestion
        return Feedback.hasDetailText(that) || (that.gradingInstruction && that.gradingInstruction.feedback);
```

File: src/main/webapp/app/entities/feedback.model.ts [hunk 2]
```diff
@@ -100,12 +100,17 @@ export class Feedback implements BaseEntity {
...
+    }
+
     /**
-     * Feedback is empty if it has 0 creits and the comment is empty.
+     * Feedback is empty if it has 0 credits and the comment is empty.
      * @param that
      */
     public static isEmpty(that: Feedback): boolean {
-        return (that.credits == undefined || that.credits === 0) && !Feedback.hasDetailText(that);
+        return (that.credits == undefined || that.credits === 0) && !Feedback.hasContent(that);
```
Comment (bassner) [inline src/main/webapp/app/entities/feedback.model.ts:113]:
```suggestion
        return !that.credits && !Feedback.hasContent(that);
```

**Review by bassner — APPROVED** [review]
Tested this and it works as described; code looks good, but I think we could simplify as shown below

**Review by bassner — APPROVED** [review]
Thank you, LGTM!

**Review by Hialus — APPROVED** [review]
Tested on TS1. 
Works as expected and code looks good to me 👍🏻

**Pull request merged**